src/test/java/picard/util/TestNGUtil.java

```diff
@@ -93,4 +94,33 @@ public static void assertEqualDoubleArrays(final double[] lhs, final double[] rh
             Assert.assertTrue(compareDoubleWithAccuracy(lhs[i], rhs[i], accuracy), "Arrays disagree at position " + i + ":  " + lhs[i] + " vs. " + rhs[i] + ". ");
         }
     }
+
+    /** A Method that returns all the Methods that are annotated with @DataProvider
+     * in a given package. Should be moved to htsjdk and used from there
+     *
+     * @param pakkage the package under hwich to look for classes and methods
+     * @return an iterator to collection of Object[]'s consisting of {Method method, Class clazz} pair.
+     * where method has the @DataProviderAnnotation and is a member of clazz.
+     */
+    public static Iterator<Object[]> getDataProviders(final String pakkage) {
+        List<Object[]> data = new ArrayList<>();
+        final ClassFinder classFinder = new ClassFinder();
+        classFinder.find(pakkage, Object.class);
+
+        for (final Class<?> testClass : classFinder.getClasses()) {
+            if (Modifier.isAbstract(testClass.getModifiers()) || Modifier.isInterface(testClass.getModifiers()))
+                continue;
+            Set<Method> methodSet = Sets.newHashSet();
+            methodSet.addAll(Arrays.asList(testClass.getDeclaredMethods()));
+            methodSet.addAll(Arrays.asList(testClass.getMethods()));
```

Comment:
I was thinking that I might want it to be run twice, so that it is run for each of the classes that inherits (e.g. MarkDuplicates testing framework)....what do you think?

Reply:
OK, I guess that makes sense, I guess the instances are different so if the method isn't static it will have a different context when run in a derived class vs the base class. (I do wonder if most of our data providers can be declared as static, and don't even need an instance to run. In the case of `MarkDuplicatesTest` the data providers could be declared static and don't depend on any inherited state, so this test is invoking them 5x when it only needs to invoke them 1x.)

Reply:
OTOH, they would be invokes several times by TestNG...so it can't be that bad that they get invoked a few more times here...can it?